### PR TITLE
review refactor: compute CtRole#forName without synonym

### PR DESCRIPTION
**Pull request by surli**
This refactor is an alternative of #1603 it just get rid of synonyms and only consider given role names. As @monperrus said it's not backward compatible but it's not a huge problem especially if it's integrated in Spoon 6.0.0

### CODE REVIEW

**Comment by monperrus** [comment]
OK for me. 
Would you add a test?

**Comment by tdurieux** [comment]
It can be a problem for me, I used it but I don't remember where ...

**Comment by surli** [comment]
> It can be a problem for me, I used it but I don't remember where ...

So, when you'll have a crash you'll know where you used it 😄

**Comment by monperrus** [comment]
I think this is a good PR that simplifies maintenance

File: src/main/java/spoon/reflect/path/CtRole.java [hunk 1]
```diff
@@ -197,11 +109,18 @@ public static CtRole fromName(String name) {
...
 	public String getCamelCaseName() {
 		String s = name().toLowerCase();
-		int i = s.indexOf("_");
-		if (i != -1) {
-			s = s.substring(0, i) + Character.toUpperCase(s.charAt(i + 1)) + s.substring(i + 2);
+		String[] tokens = s.split("_");
+		if (tokens.length == 1) {
+			return s;
+		} else {
+			StringBuffer buffer = new StringBuffer(tokens[0]);
```
Comment (msteinbeck) [inline src/main/java/spoon/reflect/path/CtRole.java]: I think you can use a StringBuilder here: https://stackoverflow.com/questions/355089/difference-between-stringbuilder-and-stringbuffer

**Pull request merged**